src/machine.rs

```diff
@@ -101,6 +106,10 @@ pub struct Evaluator<'tcx> {
     pub(crate) communicate: bool,
 
     pub(crate) file_handler: FileHandler,
+
+    /// The temporary used for storing the result of
+    /// the call to `miri_start_panic` when unwinding
+    pub(crate) panic_tmp_ptr: Option<ImmTy<'tcx, Tag>>
```

Comment:
Isn't this "the data of the exception that was thrown", to use some exception-based terminology? Unwinding doesn't happen alone, it happens with some data so that "exceptions" can be filtered and can carry information.

If that's accurate, `unwind_data` would be a better name IMO. We'd then also have a new invariant that should be stated here, namely that this is `Some` if and only if we are currently unwinding.

Reply:
Hm, you also have an `UnwindData` type though... wouldn't that type more accurately be called `CatchData` or so? It is all the arguments to `maybe_catch_panic` after all, right?